### DISCUSSION TITLE
First Time Lennox Integration User Comments/Questions

### DISCUSSION THREAD

**Discussion by 42Network** [PeteRager/lennoxs30#204] [General]
Hello. I am very pleased to find this excellent integration. Our 25-year-old dual up and down 2.5/3 ton systems finally died a few weeks ago and we now have a brand new installation of a three-zone 5-ton variable speed AC two stage heat Lennox system. 

OU: SL28XCV-060
IU: SL280UH090V60C
Zone: iComfort DCM
Thermostat: S30
Zone sensors: 17A30

While I am still staggering under the cost of having something the price of a minivan in my attic, we are loving it so far.

Some questions, though, about this integration.

1. After I enabled diagnostics on the integration config screen, I found the diagnostics entities were all created, but were unavailable.
2. The diagnostic entities did become available after I set "system_diagnostic_level" to 1 by hand in HA. Is this normal that the diags aren't populated until we manually change the diag level?
3.  Even though your documentation says the diag level should be 2, I get diag data with it set to 1 including AC Inverter Power. What is the actual difference between 1 and 2?
4. Does the diag level persist across Lennox system restarts? I've had to once or twice put system_diagnostic_level back to 1 and I think that was after I had "rebooted" the Lennox system.
5. Is there a way to make the diag level permanently persistent? I.e. I always want to have heat/cool demand and CFM demand on my dashboards.
6. I still haven't blocked my S30 from the Internet at my firewall but I have not noticed any unusual restarts of my Lennox system after running for a week. Is the crashing due to diagnostics random? Has it been fixed in the newest firmware I likely have?
7. If I do have to block my S30, is it possible to still allow it to access weather data for the screen saver?
8. I see that the S30 has port 22 open in addition to 443. Has anyone found a way to use SSH?
9. My zone controller has a USB port inside and what appears to be an SD memory card. Has anyone explored these items?
10. Is it possible to "snoop" the RS-BUS protocol between the communicating devices using a RS-485/USB dongle attached to a computer?

I know that was a lot of questions. I really appreciate the HA community and in particular all the work that went into this Lennox integration. I will find the donation link and put some coin there. And I will be very willing to send any logs or participate in any testing needed, particularly if there are things you want to try on my latest hardware from Lennox.

Thanks/Nathan

**Comment by PeteRager** [comment]
Thanks and great questions

1+2 - the S30 only sends diagnostic data when the diag level is non-zero.  So yes, when the diag level is zero those entities are unavailable.
3 - we don’t know.  What we do know is the Lennox technician app sets it to 2 and we have a number of users with stable systems running at level 2.  Hence the recommendation is 2.
4+5  The S30 does not persist the diag level so yes when it restarts it goes back to zero.   I’ve seen this a few times, it’s not ideal. Let’s open an issue.  The solution may be a simple HA automation.  Either way the integration should document and provide a solve.
6. I think this is a complex interaction of internet speed and using the mobile app.  We have had users with very unstable systems due to diags being on with internet connected.  If it’s working for you, super.  I would recommend to monitor the sysuptime attribute to look for resets.
7. Good question.   I don’t know.  If it’s going directly to the weather provider URL, then yes you may be able to configure your router to allow this. Open an issue if this is important to figure out.  
8. Have not been able to access ssh.  My conclusion was, I think they left the port open by mistake and there are no ssh users / keys configured on the system.  Happy to share those notes if you want to pursue further. 
9. My system does not have a zone controller and haven’t heard of anyone connecting into it.  We have found a way to directly control the CFM per zone in real-time using the parameters beta. That said, Lennox does provide detailed technical docs that may have the answers. 
10. I haven’t found any docs on RSBus, though there are a couple of Lennox patents filed.  There is an port 443 interface to send RSBus commands, but since I can’t find any info on valid commands I haven’t tried it.   Did you find something that said it’s RS-485?

**Comment by PeteRager** [comment]
Had a power flicker last night, reset the S30 and then this morning notices diags were turned off, so I think that problem needs solving.  An approach is to create a automation that detects that the sysuptime has reset, meaning the new value is less than the old value, then set the diagLevel back to 2.

**Comment by 42Network** [comment]
This (older) S30 tips document seems to imply it uses RS-485. I'm not sure if this is still true. 

https://www.scribd.com/document/555169180/LENNOX-iComfort-S30-Troubleshooting-Tips

But RS-485/USB dongles are pretty cheap and I may try it myself. 

https://www.amazon.com/Adapter-Serial-Terminal-Ferrite-Windows/dp/B08SM651R9/ref=sr_1_2?keywords=rs-485%2Busb&qid=1663708524&sr=8-2&th=1

**Reply by PeteRager** [reply]
Interesting.  There’s always a risk a smoking the unit if it’s not or the pin out is non standard.   For the thermostat panel, the cable is low delivering power.
